agent/ebs/watcher.go

```diff
@@ -112,6 +134,28 @@ func (w *EBSWatcher) HandleEBSResourceAttachment(ebs *apiebs.ResourceAttachment)
 		})
 	}
 
+	// start EBS CSI Driver Managed Daemon
+	if w.taskEngine.GetEbsDaemonTask() != nil {
+		log.Infof("engine ebs CSI driver is running with taskID: %v", w.taskEngine.GetEbsDaemonTask().GetID())
+	} else {
+		if ebsCsiDaemonManager, ok := w.taskEngine.GetDaemonManagers()["ebs-csi-driver"]; ok {
+			if csiTask, err := ebsCsiDaemonManager.CreateDaemonTask(); err != nil {
+				// fail attachment and return
+				log.Errorf("Unable to start ebsCsiDaemon for task %s in the engine: error: %s", csiTask.GetID(), err)
+				csiTask.SetKnownStatus(apitaskstatus.TaskStopped)
+				csiTask.SetDesiredStatus(apitaskstatus.TaskStopped)
+				w.taskEngine.EmitTaskEvent(csiTask, err.Error())
```

Comment:
For my information, does this send an event to the backend?

Reply:
messages on the engine's task events channel are consumed at the other end by task/attachment message handlers.  This event ends up making a call to ecs `submit-attachment-state-changes` where an attachment looks like 
{ 
         "[attachmentArn](https://docs.aws.amazon.com/AmazonECS/latest/APIReference/API_AttachmentStateChange.html#ECS-Type-AttachmentStateChange-attachmentArn)": "string",
         "[status](https://docs.aws.amazon.com/AmazonECS/latest/APIReference/API_AttachmentStateChange.html#ECS-Type-AttachmentStateChange-status)": "string"
      }     
https://docs.aws.amazon.com/AmazonECS/latest/APIReference/API_SubmitAttachmentStateChanges.html

Reply:
Managed daemon tasks would be internal, right? IIUC `EmitTaskEvent` does not send an event for internal tasks as the `api.NewTaskStateChangeEvent(task, reason)` returns a `ErrShouldNotSendEvent{task.Arn}` error.

https://github.com/aws/amazon-ecs-agent/blob/e446222b214ce790f4af7b7169342fce67ecf707/agent/engine/docker_task_engine.go#L970C2-L970C57

https://github.com/aws/amazon-ecs-agent/blob/e446222b214ce790f4af7b7169342fce67ecf707/agent/api/statechange.go#L117-L119

Am I missing something? How will `EmitTaskEvent` call result in an attachment change event?